src/client/authorization-code.ts

```diff
@@ -81,11 +93,43 @@ export class OAuth2AuthorizationCodeClient {
       code: params.code,
       redirect_uri: this.redirectUri,
       client_id: this.client.settings.clientId,
-      code_verifier: params.codeVerifier,
+      code_verifier: this.codeVerifier,
     };
     return tokenResponseToOAuth2Token(this.client.request('tokenEndpoint', body));
 
   }
 
 
 }
+
+export function getCodeVerifier(): string {
+  const arr = new Uint8Array(32);
+  crypto.getRandomValues(arr);
+  return base64Url(arr);
```

Comment:
Working with buffers always jostles me a tad, but couldn't you just encode directly from the buffer `arr.toString('base64url')`? https://nodejs.org/api/buffer.html#buffers-and-character-encodings

Reply:
I try to write everything using APIs that exist in browsers. The moment we use Buffer webpack includes a huge package. So, if you know of a good way of doing this without relying on node-exclusive APIs, tell me!

Reply:
ah ok makes sense